app/controllers/buyers/applications_controller.rb

```diff
@@ -191,8 +187,18 @@ def find_provider
     @provider = current_account
   end
 
-  def find_application_plan
-    @application_plan = @provider.application_plans.find params[:cinstance][:plan_id]
+  def accessible_services
+    @accessible_services ||= current_user.accessible_services.includes(:application_plans)
+  end
+
+  helper_method :accessible_services
+
+  def accessible_plans
+    current_account.application_plans.where(issuer: accessible_services)
+  end
+
+  def plan_id
+    @plan_id ||= params.require(:cinstance).permit(:plan_id).tap { |plan_params| plan_params.require(:plan_id) }[:plan_id]
```

Comment:
so how the params should look like? `cintances[:plan_id][:plan_id]` ?

Reply:
`{cinstance: {plan_id: 1}}` for example.
Look at the test 😄 but it is exactly the same format as before, just that now the param is mandatory and otherwise raises the proper error instead of trying to find by nil.
look at https://github.com/3scale/porta/pull/1798#discussion_r408688711

Reply:
Why not just `params.require(:cinstance).require(:plan_id)`?

Reply:
That doesn't work @hallelujah . It always needs a permit at the end of the chain.
Look at [the documentation of ActionController::Parameters#require](https://api.rubyonrails.org/classes/ActionController/Parameters.html#method-i-require)
![image](https://user-images.githubusercontent.com/11318903/85267265-521c9680-b475-11ea-93b6-79273e522735.png)

Reply:
I do not get it, why is it safer?

Reply:
OK got it, imagining people sending a has as `:plan_id`

```ruby

params = ActionController::Parameters.new(person: {name: {a: :b}})
params.require(:person).require(:name)
# => <ActionController::Parameters {"a" => "b"} permitted: false>

params.require(:person).permit(:name)
# => <ActionController::Parameters {} permitted: false>
```